@here/harp-vectortile-datasource/lib/VectorTileDecoder.ts

```diff
@@ -111,149 +113,145 @@ export class VectorTileDataProcessor implements IGeometryProcessor {
-                this.m_featureModifiers.find(fm => {
-                    // TODO: The logic of feature ignore should be actually in the feature filtering
-                    // mechanism - see OmvFeatureFilter.
-                    return !fm.doProcessPointFeature(layer, env, storageLevel);
-                }) !== undefined
-            ) {
-                return;
-            }
-        }
-
-        const techniques = this.applyKindFilter(
-            this.m_styleSetEvaluator.getMatchingTechniques(env, layer, "point"),
-            GeometryKind.Label
+        assert(this.m_decodedTileEmitter !== undefined);
+        // Pass feature modifier method to processFeature if there's any modifier. Get it from any
+        // modifier, processFeature will later apply it to all using Function.apply().
+        const featureModifierFunc = this.m_featureModifiers?.[0].doProcessPointFeature;
+        this.processFeature(
+            layer,
+            extents,
+            geometry,
+            "point",
+            properties,
+            featureId,
+            this.m_decodedTileEmitter!.processPointFeature,
+            GeometryKind.Label,
+            featureModifierFunc
         );
-
-        if (techniques.length === 0) {
-            if (this.m_showMissingTechniques) {
-                logger.log(
-                    "OmvDecoder#processPointFeature: no techniques for object:",
-                    JSON.stringify(env.unmap())
-                );
-            }
-            return;
-        }
-        const context: AttrEvaluationContext = {
-            env,
-            cachedExprResults: new Map()
-        };
-
-        if (this.m_decodedTileEmitter) {
-            this.m_decodedTileEmitter.processPointFeature(
-                layer,
-                extents,
-                geometry,
-                context,
-                techniques
-            );
-        }
     }
 
     /** @override */
     processLineFeature(
         layer: string,
         extents: number,
         geometry: ILineGeometry[],
-        env: MapEnv,
-        storageLevel: number
+        properties: ValueMap,
+        featureId: string | number | undefined
     ): void {
-        if (this.m_featureModifiers !== undefined) {
-            if (
-                this.m_featureModifiers.find(fm => {
-                    return !fm.doProcessLineFeature(layer, env, storageLevel);
-                }) !== undefined
-            ) {
-                return;
-            }
-        }
-
-        const techniques = this.applyKindFilter(
-            this.m_styleSetEvaluator.getMatchingTechniques(env, layer, "line"),
-            GeometryKind.Line
+        assert(this.m_decodedTileEmitter !== undefined);
+        // Pass feature modifier method to processFeature if there's any modifier. Get it from any
+        // modifier, processFeature will later apply it to all using Function.apply().
+        const featureModifierFunc = this.m_featureModifiers?.[0].doProcessLineFeature;
```

Comment:
Why do you use just the first feature modifier function? It is an array so there could be multiple ones?

Reply:
It explains in the comment above, to get a method you need to get it from any instance of the class, it doesn't matter which. Later I'm binding that method to different objects using Function.apply(). 
Anyway, I've refactored the code and now that's not there anymore. I've moved the code to OmvDataAdapter.